gcc/rust/expand/rust-macro-builtins.cc

```diff
@@ -0,0 +1,27 @@
+#include "rust-macro-builtins.h"
+#include "rust-diagnostics.h"
+#include "rust-expr.h"
+#include "rust-session-manager.h"
+
+namespace Rust {
+AST::ASTFragment
+MacroBuiltin::assert (Location invoc_locus, AST::MacroInvocData &invoc)
+{
+  rust_debug ("assert!() called");
+
+  return AST::ASTFragment::create_empty ();
```

Comment:
this could call GCC's builtin assert but we need a way to preserve that information into the HIR that its a builtin call. Maybe we could introduce a builtin call HIR node and handle it that way.

Reply:
Yeah I agree. This was just to make sure that the builtin was working. I'm working on actually implementing assert in my `add-assert-macro` branch [here](https://github.com/CohenArthur/gccrs/blob/add-assert-macro/gcc/rust/expand/rust-macro-builtins.cc#L56). I don't know how easy it would be to insert a call to __builtin_assert at the AST level?